### PR TITLE
Relocate existing k8s/terraform infrastructure

### PR DESCRIPTION
**Pull request by brews**
This PR migrates two bits of existing infrastructure. Moving this logic here will make our dependence on complex infrastructure stuff more transparent and open.

The first addition is the Kubernetes infrastructure from https://github.com/ClimateImpactLab/downscaleCMIP6-infra-argo. This is moved to a new `/infrastructure/kubernetes/` directory in this repo.

This infra is automatically watched/managed/updated by an ArgoCD install on the cluster.

A README.md at the base of that directory describes what's what and should be readable by anyone familiar with Kubernetes and basic ArgoCD.

The other addition is the terraform deploying. This is migrated from https://github.com/ClimateImpactLab/downscaleCMIP6-infra to `/infrastructure/terraform/` in this repo. This code sets up the hard hardware, including the clusters deployed to in `/infrastructure/kubernetes/`.

Changes to Terraform will be planned automatically through CI/CD, though viewing the plans and applying changes requires manual, external confirmation.

### CODE REVIEW

**Comment by brews** [comment]
I just added the Terraform infra. I'll need to try to set up CI/CD for it after this merges.

File: workflows/notes.sh [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+    -n argo --watch
+
+
+#   File "/argo/staging/script", line 26, in <module>
+#     QDMdg = sdba.adjustment.QuantileDeltaMapping.from_dataset(qdm_model_ds)
+#   File "/opt/conda/lib/python3.8/site-packages/xclim/sdba/base.py", line 125, in from_dataset
+#     obj = cls.from_string(defstr)
+#   File "/opt/conda/lib/python3.8/site-packages/xclim/sdba/base.py", line 82, in from_string
+#     params[dd["name"]] = cls._parametrizable_classes[
+# KeyError: 'Grouper'
```
Comment (dgergel) [inline workflows/notes.sh:26]: do we need this bit of commented out code?

File: workflows/notes.sh [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+# dc6-dev-plrg4-580946482:     return empty(store=self._store, path=path, chunk_store=self._chunk_store,
+# dc6-dev-plrg4-580946482:   File "/opt/conda/lib/python3.8/site-packages/zarr/creation.py", line 227, in empty
+# dc6-dev-plrg4-580946482:     return create(shape=shape, fill_value=None, **kwargs)
+# dc6-dev-plrg4-580946482:   File "/opt/conda/lib/python3.8/site-packages/zarr/creation.py", line 121, in create
+# dc6-dev-plrg4-580946482:     init_array(store, shape=shape, chunks=chunks, dtype=dtype, compressor=compressor,
+# dc6-dev-plrg4-580946482:   File "/opt/conda/lib/python3.8/site-packages/zarr/storage.py", line 348, in init_array
+# dc6-dev-plrg4-580946482:     _init_array_metadata(store, shape=shape, chunks=chunks, dtype=dtype,
+# dc6-dev-plrg4-580946482:   File "/opt/conda/lib/python3.8/site-packages/zarr/storage.py", line 377, in _init_array_metadata
+# dc6-dev-plrg4-580946482:     raise ContainsArrayError(path)
+# dc6-dev-plrg4-580946482: zarr.errors.ContainsArrayError: path 'lat' contains an array
```
Comment (dgergel) [inline workflows/notes.sh:63]: should we remove this too?

**Review by dgergel — CHANGES_REQUESTED** [review]
@brews this looks so awesome, thank you for setting this all up and migrating it to this repo! I have a few comments, and I was wondering if we want to take out some of the errors that are commented out but appear in the `notes` section? Or did you want to have a record of these? Also was wondering what the point was of this file `infrastructure/terraform/outputs.tf` since it's empty? I'm super green with terraform though so maybe that has a purpose I'm not aware of...

File: workflows/notes.sh (hunk 1 above)
Comment (brews) [reply workflows/notes.sh]: Ah, no. Thanks for catching that. The entire file should be removed.

(done)

File: workflows/notes.sh (hunk 2 above)
Comment (brews) [reply workflows/notes.sh]: Yeah, thanks. I'm removing this entire file.

Marking resolved.

File: infrastructure/kubernetes/README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
+# downscaleCMIP6-infra-argo
+Kuberenetes internals and Argo Workflow configuration for CMIP6 downscaling project.
```
[the remark this answers is not in the document]
Comment (brews) [reply infrastructure/kubernetes/README.md]: Got it. Thanks!

Marking resolved.

File: infrastructure/kubernetes/README.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+Manifests for Argo Workflow itself are in `./argo/`. The `./workflows-default/` directory contains manifests required to run analysis Workflows. `./minio/` is a configuration Helm Chart to deploy the artifact repository. Cluster internals are mostly managed with `argocd`. The hard infrastructure we're deploying on top of is in `/infrastructure/terraform/`.
+
+## Updating
+
+If there is already has a deployment -- file a pull request with changes to `main`. Merged PRs are checked and automatically pulled.
```
[the remark this answers is not in the document]
Comment (brews) [reply infrastructure/kubernetes/README.md]: I updated it to make it more clear (and grammatically correct) — I hope.

Please let me know if you'd like further changes, @dgergel.

**Comment by brews** [comment]
> @brews this looks so awesome, thank you for setting this all up and migrating it to this repo! I have a few comments, and I was wondering if we want to take out some of the errors that are commented out but appear in the `notes` section? Or did you want to have a record of these? Also was wondering what the point was of this file `infrastructure/terraform/outputs.tf` since it's empty? I'm super green with terraform though so maybe that has a purpose I'm not aware of...

Thanks for the feedback, @dgergel. This is ready for a re-review.

The `notes.sh` file is old scratch notes. It shouldn't be here. I removed it.

The outputs.tf is for denoting output variables from a Terrafrom workspace. I removed it. I can drop it back in here if we have need for it again in the future.

Please let me know if you have additional comments or changes. Thank you!

**Review by dgergel — APPROVED** [review]
Looks great, thanks @brews!

**Pull request merged**